src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py

```diff
@@ -510,50 +615,106 @@ def add_data_set(self, key=""):
+        attrs = value.attrs
+        signal = attrs.get("signal", "I")
+        i_axes = attrs.get("I_axes", ["Q"])
+        q_indices = attrs.get("Q_indices", [0])
+        q_indices = map(int, self.check_is_list_or_array(q_indices))
+        i_axes = self.check_is_list_or_array(i_axes)
+        keys = value.keys()
+        # Assign attributes to appropriate class variables
+        self.mask_name = attrs.get("mask")
+        for val in q_indices:
+            self.q_names.append(i_axes[val])
+        self.i_name = signal
+        self.i_node = value.get(self.i_name)
+        for item in self.q_names:
+            if item in keys:
+                q_vals = value.get(item)
+                if q_vals.attrs.get("uncertainties") is not None:
+                    self.q_uncertainty_names = q_vals.attrs.get("uncertainties")
+                elif q_vals.attrs.get("uncertainty") is not None:
+                    self.q_uncertainty_names = q_vals.attrs.get("uncertainty")
+                if isinstance(self.q_uncertainty_names, basestring):
+                    self.q_uncertainty_names = self.q_uncertainty_names.split(",")
+                if q_vals.attrs.get("resolutions") is not None:
+                    self.q_resolution_names = q_vals.attrs.get("resolutions")
+                if isinstance(self.q_resolution_names, basestring):
+                    self.q_resolution_names = self.q_resolution_names.split(",")
+        if self.i_name in keys:
+            i_vals = value.get(self.i_name)
+            self.i_uncertainties_name = i_vals.attrs.get("uncertainties")
+            if self.i_uncertainties_name is None:
+                self.i_uncertainties_name = i_vals.attrs.get("uncertainty")
+
+    def _is2d(self, value, i_base="", q_base=[]):
+        """
+        A private class to determine if the data set is 1d or 2d.
 
-        :param parent_list: List of parents nodes in the HDF5 file
+        :param value: Nexus/NXcanSAS data group
         :param basename: Approximate name of an entry to search for
-        :return:
+        :return: True if 2D, otherwise false
         """
-
-        entry = False
-        key_prog = re.compile(basename)
-        top = self.raw_data
-        for parent in parent_list:
-            top = top.get(parent)
-        for key in top.keys():
-            if key_prog.match(key):
-                entry = True
-                break
-        return entry
+        i_basename = i_base if i_base != "" else self.i_name
+        i_vals = value.get(i_basename)
+        q_basename = q_base if q_base != [] else self.q_names
+        q_vals = value.get(q_basename[0])
+        self.multi_frame = True if (i_vals is not None and q_vals is not None
+                                    and len(i_vals.shape) != 1
+                                    and len(q_vals.shape) == 1) else False
+        return (i_vals is not None and i_vals.shape is not None
+                and len(i_vals.shape) != 1 and not self.multi_frame)
```

Comment:
If `i_vals` is an `ndarray` then it has a shape which is never `None`, so don't need the second condition.